### ISSUE TITLE
Flow mismatch between autosized PTHP and hard value in variable speed DX coils child object results incorrect operation

### ISSUE THREAD

**Issue by nealkruis** [NREL/EnergyPlus#5251]
I've created a defect file to illustrate this (Unitary_and_PTHP.idf). In this file there is a central airloop with a unitary system serving a single zone with a PTHP. The `Output:Variables` illustrate that the total air into the zone and the total air out of the zone are far from balancing.
- There is never any return air even though the ATU is supplying air to the space on occasion. 
- The PTHP is always drawing more air out of the zone than it is supplying (regardless of the ATU operation).

Despite the air balance issue, the file still runs to completion without warnings or errors related to this problem.

The systems are also completely unable to control zone temperature even though they are given adequate capacity. I'm not sure if this is a related problem or something entirely different.

**Comment by Nigusse** [comment]
I investigated this bug and the finding is as follows:

The mass balance issue is caused by flow mismatch between the PTHP parent autosized flow rate and the hard-sized rated flow rates in the variable Speed DX cooling/heating coils. In the defect file specified the DX coils lowest speed rated air flow rate is higher than the parent (PTHP) autosized air flow rate. The supply fan flow rate is also autosized and is the same as the parent. I modified to defect file and here are my findings:

Test 1: modified the DX coils nominal flow rates and nominal capacity to autosized and scaled down the rated capacity and flow rates of the DX coils for each speed according to the ratios 0.25, 0.5, .75, 1.0 for speed 1, 2, 3, and 4, respectively. The zone mass flow were balanced and the zone temperature set point were met. Only the PTHP was able to meet the load.

Test 2: modified the DX coils nominal flow rates and nominal capacity to autosized but kept the original defect file hard-sized capacities and flow rates for each speed. This test case produced the similar zone flow unbalance for the PTHP and the zone thermostat conditions were not met. Do we allow the AirLoop ATU zone object to be second and a PTHP first in zone control sequence?

Test 3: Reversed the control sequence in the ZoneHVAC:EquipmentList, the ATU first and the PTHP second. Airloo only was able to meet the load for almost the entire year and only a handful time the PTHP was on at small flow rates and there was difference between the supply and exhaust air flow rates for the PTHP. The AirLoop flow rates were always balanced. 

Test 4: replaced the variable speed PTHP with single speed PTHP.  The zone flow was balanced and the zone conditions were met.  The supply air flow rate and return air flow rate were identical even when the DX coils were hard-sized higher flow rate than the parent.

Test 5: Removed the AirLoop such that the zone is served with the PTHP alone and I found the same flow imbalance and the PTHP failed to meet the zone conditions. This condludes the problem is in the PTHP with variable speed DX coils.

Based on these tests and stepping through the code shows that the Variable Speed PTHP simulates the DX coils not at the parent flow rate or at least does not proportionally scale down to flows that match the parent and for that reason the zone exhaust air flow rate shows erratic behavior and mismatched to the supply air flow rate. The variable speed DX coils used in the PTHP should not be allowed to run at a higher flow rate than the parent.  Also the section that checks if the fan flow is less than the DX coils rated flow rate is not executed hence no warning is provided.

I suggest that the title of this bug is modified to "Flow mismatch between autosized PTHP and hard sized variable speed DX coils child object results incorrect operation of the PTHP" and point the bug to the developer.  @mjwitte @Myoldmopar @nealkruis ?

**Comment by Nigusse** [comment]
@shenb Can you look into this?

**Comment by shenb** [comment]
Sure, I will look into it. It may take some time.

Thanks,

Bo

**Comment by rraustad** [comment]
See https://github.com/NREL/EnergyPlus/issues/5430 before attempting to correct this.

**Comment by rraustad** [comment]
PR #5632 fixes this issue. Ran defect file using priority 1 and 2 for zone wshp wrt the air loop TU. No imbalance throughout the entire year.

**Comment by rraustad** [comment]
I was thinking after the call that this could be fixed simply by fixing the 2 lines in Init to get the flows right:

```
PTUnit( PTUnitNum ).HeatMassFlowRate( Iter ) = VarSpeedCoil( PTUnit( PTUnitNum ).DXHeatCoilIndex ).MSRatedAirMassFlowRate( Iter ) * MulSpeedFlowScale;
```

**Comment by mjwitte** [comment]
Closed via #5632
